runtime/test/test_tensor.cpp

```diff
@@ -0,0 +1,59 @@
+#include <cstdlib>
+#include <iostream>
+#include "blueoil.hpp"
+
+int test_tensor() {
+    float tensor_data[][3] = {
+			      {1, 2, 3},
+			      {7, 8, 9}
+    };
+    float tensor_data2[][2] = {  // equals data, different shape.
+			       {1, 2},
+			       {3, 7},
+			       {8, 9}
+    };
+    float tensor_data3[][3] = {  // equals shape, different data.
+			       {1, 2, 3},
+			       {7, 0, 9}
+    };
+    blueoil::Tensor tensor0({2, 3}, (float *) tensor_data);
+    blueoil::Tensor tensor1({2, 3}, (float *) tensor_data);
+    blueoil::Tensor tensor2({3, 2}, (float *) tensor_data2);  // shape diff
+    blueoil::Tensor tensor3({2, 3}, (float *) tensor_data3);  // data diff
```

Comment:
could we use C++ casts instead of C ones.
- `reinterpret_cast` for pointers
- `static_cast` for float -> int, int -> float, etc